packages/orderbook/src/orderbook.ts

```diff
@@ -622,28 +701,42 @@ export class Orderbook {
     orderIds: string[],
     accountAddress: string,
   ): Promise<CancelOrdersOnChainResponse> {
-    const orderResults = await Promise.all(orderIds.map((id) => this.apiClient.getListing(id)));
+    const listingResultsPromises = Promise.all(
+      orderIds.map((id) => this.apiClient.getListing(id).catch((e: ApiError) => {
+        if (e.status === 404) {
+          return undefined;
```

Comment:
is this something we've always done specifically for 404's?

Reply:
When we're given an order ID, we don't know what order type it refers to. To retrieve the order, we hit each of the endpoints (`/listings`, `/bids'`) to retrieve the order. Given an order will only be of one order type, it will 404 for all but one of the endpoints. Further below we then check that all orders were found. This is an incredibly naive approach for now. The thinking is on-chain cancellation via the SDK isn't something we'll see often. We can invest in a less naive implementation should this assumption change.